### PR TITLE
cmake file changes for macOS universal2 support

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

"Universal2" allows us  creating a single binary that runs natively on both Apple silicon and Intel-based Mac computers. There are two ways of doing this:
1.  Build for the two archs at the same time with "-arch arm64 -arch x86_64". 
2.  Build two binaries separately, then use a tool to combine them together after the build. 

The second one already works. The first one is what I'm providing.  You can try the new thing in a way like:
```
python3 tools/ci_build/build.py --skip_submodule_sync --build_dir b --cmake_extra_defines 'CMAKE_OSX_ARCHITECTURES=x86_64;arm64' --parallel
```

To find more information about Universal2, please
https://developer.apple.com/documentation/apple-silicon/building-a-universal-macos-binary

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by snnn** [comment]
This PR is split from #8754

File: cmake/onnxruntime_common.cmake [hunk 1]
```diff
@@ -185,18 +167,11 @@ if(MSVC)
...
-    set(ARM64 TRUE)
-  elseif (CMAKE_OSX_ARCHITECTURES STREQUAL "arm")
-    set(ARM TRUE)
-  elseif (CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64")
-    set(X86_64 TRUE)
-  elseif (CMAKE_OSX_ARCHITECTURES STREQUAL "i386")
-    set(X86 TRUE)
+elseif(APPLE)
+  if(CMAKE_OSX_ARCHITECTURES_LEN LESS_EQUAL 1)
+    set(X64 TRUE)
```
Comment (gwang-msft) [inline cmake/onnxruntime_common.cmake:172]: Is this right? What if I specify is to arm64 only, then x64 is true?

**Comment by gwang-msft** [comment]
Can you also disable this when the `onnxruntime_BUILD_APPLE_FRAMEWORK` is set to true? We are stitching static libs together, however, it won't work for universal lib

File: cmake/onnxruntime_common.cmake (hunk 1 above)
Comment (snnn) [reply cmake/onnxruntime_common.cmake:172]: Yes, it is only used at line 206 this file.

**Comment by snnn** [comment]
> Can you also disable this when the `onnxruntime_BUILD_APPLE_FRAMEWORK` is set to true? We are stitching static libs together, however, it won't work for universal lib

Disable what? Sorry I lose the context.

**Comment by gwang-msft** [comment]
> > Can you also disable this when the `onnxruntime_BUILD_APPLE_FRAMEWORK` is set to true? We are stitching static libs together, however, it won't work for universal lib
> 
> Disable what? Sorry I lose the context.

If we try to build universal binary when `onnxruntime_BUILD_APPLE_FRAMEWORK` is true, the output static framework will not be usable. since we stitch multiple static libraries together, but that would not work for universal static libraries

File: cmake/onnxruntime_common.cmake (hunk 1 above)
Comment (gwang-msft) [reply cmake/onnxruntime_common.cmake:172]: This `x64` name is confusing

**Review by gwang-msft — APPROVED** [review]

**Comment by RajalakshmiSR** [comment]
@snnn This affects POWER10 build. `mlas_common_srcs`   in cmake/onnxruntime_mlas.cmake is used to add some build option . I will create a PR to address this.

**Pull request merged**